### PR TITLE
Logging tools and frontend

### PR DESCRIPTION
**Pull request by zxan1285**

### CODE REVIEW

File: server/lib/processLogs.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      return next();
+    }
+
+    const blobService = azure.createBlobService(config('STORAGE_ACCOUNT_NAME'), config('STORAGE_ACCOUNT_KEY'));
+
+    const remapLogs = (record) => {
+      record.type_code = record.type;
+      record.type = loggingTools.logTypes.get(record.type);
+
+      let agent = useragent.parse(record.user_agent);
```
Comment (sandrinodm) [inline server/lib/processLogs.js:26]: Make sure user agent is not null

File: server/lib/processLogs.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      record.device_version = agent.device.toVersion();
+      return record;
+    };
+
+    const onLogsReceived = (logs, callback) => {
+      if (!logs || !logs.length) {
+        return callback();
+      }
+
+      logger.info(`Sending ${logs.length} logs to Azure.`);
```
Comment (sandrinodm) [inline server/lib/processLogs.js:39]: Azure Blob Storage

File: server/lib/processLogs.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+        logger.info('Upload complete.');
+        return callback();
+      });
+    };
+
+    const slack = new loggingTools.reporters.SlackReporter({
+      hook: config('SLACK_INCOMING_WEBHOOK_URL'),
+      username: 'auth0-logs-to-azure-blob-storage',
+      title: 'Logs To Azure' });
```
Comment (sandrinodm) [inline server/lib/processLogs.js:59]: Azure Blob Storage

File: server/index.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  app.use(processLogs(storage));
+
+  app.use(expressTools.routes.dashboardAdmins({
+    secret: config('EXTENSION_SECRET'),
+    audience: 'urn:logs-to-azure-blob-storage',
+    rta: config('AUTH0_RTA').replace('https://', ''),
+    domain: config('AUTH0_DOMAIN'),
+    baseUrl: config('PUBLIC_WT_URL') || config('WT_URL'),
+    clientName: 'Logs to Azure',
```
Comment (sandrinodm) [inline server/index.js:52]: Azure Blob Storage

File: webtask.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const tools = require('auth0-extension-express-tools');
+
+const expressApp = require('./server');
+const config = require('./server/lib/config');
+const logger = require('./server/lib/logger');
+
+const createServer = tools.createServer((config, storage) => {
+  logger.info('Starting Auth0 Logs to Azure Extension - Version:', process.env.CLIENT_VERSION);
```
Comment (sandrinodm) [inline webtask.js:8]: Azure Blob Storage

**Review by sandrinodimattia — APPROVED** [review]

**Pull request merged**